public/app/features/explore/extensions/AddToDashboard/addToDashboard.ts

```diff
@@ -64,39 +63,18 @@ function getLogsTableTransformations(panelType: string, options: AddPanelToDashb
   return transformations;
 }
 
-export async function setDashboardInLocalStorage(options: AddPanelToDashboardOptions) {
+export function buildDashboardPanelFromExploreState(options: ExploreToDashboardPanelOptions): Panel {
   const panelType = getPanelType(options.queries, options.queryResponse, options?.panelState);
 
-  const panel = {
+  return {
+    //@ts-ignore
```

Comment:
Could we fix type returned by `getPanelType` instead of ignoring all types? I think [this line](https://github.com/grafana/grafana/blob/5418a95bdee9f6acd647ea9ef42d643dad003ee9/public/app/features/explore/extensions/AddToDashboard/addToDashboard.ts#L106) is problematic  returning more than `string`:

<img width="1153" alt="Screenshot 2024-10-11 at 12 43 33" src="https://github.com/user-attachments/assets/ee33fb84-411c-4c7a-acc5-d621dc500ea5">

Reply:
@ifrost have not changed that line, it's strange though,. preferredVisualisationPluginId is typed string. So do not get any type error on that line.